Dockerfile

```diff
@@ -1,10 +1,20 @@
 FROM python:3.9
 
-COPY ./requirements.txt /app/requirements.txt
-RUN pip install -r /app/requirements.txt
+RUN pip install -U pip \
+    && apt-get update \
+    && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+
+ENV PATH="${PATH}:/root/.poetry/bin"
 
 COPY . /app
 WORKDIR /app/
 
-ENTRYPOINT ["./run.sh"]
-CMD []
+RUN poetry install
+
+# easy bruh...
+#ENTRYPOINT ["./run.sh"]
+
+# easter eggs 😝
+RUN echo "PS1='🐳:\[\033[1;36m\]\h \[\033[1;34m\]\W\[\033[0;35m\]\[\033[1;36m\]$ \[\033[0m\]'" >> ~/.bashrc
+
+CMD /bin/bash
```

Comment:
does this hold it open? I was pretty sure this still dies immediately with `docker compose up` which is why the loop is required

Reply:
big fan if so, it's definitely better than a bash while loop

Reply:
The cmd line just runs that command when you enter, similar to specifying the same but through exec it. Added a tty arg to the compose solves the persistence.  Image below shows a few tests.  TTL is a clean solution.

![Screen_Shot_2021-07-23_at_6 33 57_PM](https://user-images.githubusercontent.com/8229664/126854084-88e66c9a-e7a2-40ed-bbdb-9c9fd1d1d787.png)